sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/TestUtils.java

```diff
@@ -84,10 +96,48 @@
      * @return The namespace connection string.
      */
     public static String getConnectionString(boolean withSas) {
+        String connectionString = getPropertyValue("AZURE_SERVICEBUS_NAMESPACE_CONNECTION_STRING");
```

Comment:
Is there a reason we're copying this entire logic instead of using the ServiceBusSharedKeyCredential and generating the signature from that?

Reply:
The connection string of `ServiceBus` has `Endpoint=...;SharedAccessKeyName=...;SharedAccessKey=...` and `Endpoint=...;SharedAccessKeyName=...;SharedAccessKey=...;EntityPath=...` these two formats, in order to the extensibility of test cases and the generality of the generated sas connection string, so copying this entire logic.